frontend/src/components/HousingLocatorTable.tsx

```diff
@@ -0,0 +1,312 @@
+    setPopup(true);
+    demoteUser(user)
+      .then((value) => {
+        if (value.success) {
+          dataContext.refetchData();
+        }
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  return (
+    <HLTableWrapper>
+      <HLTableContainer>
+        <HLTableHeader>Name</HLTableHeader>
+        {dataContext.allHousingLocators
+          .slice((pageNumber - 1) * ENTRIES_PER_PAGE, pageNumber * ENTRIES_PER_PAGE)
+          .map((locator: User, index: number) => (
+            <HLRow key={index}>
+              {locator.firstName} {locator.lastName}
+              <DeleteIcon
+                src="trash-can.svg"
+                onClick={() => {
+                  if (locator !== null) {
+                    setSelectedUser(locator);
+                    setPopup(true);
+                  }
+                }}
+              ></DeleteIcon>
+            </HLRow>
+          ))}
+      </HLTableContainer>
+      {popup && selectedUser && (
+        <>
+          <Overlay />
+          <Modal>
+            <XWrapper>
+              <XButton
+                onClick={() => {
+                  setPopup(false);
+                }}
+              >
+                &times;
+              </XButton>
+            </XWrapper>
+            <HeadingWrapper>Remove Housing Locator</HeadingWrapper>
+            <WarningMessageWrapper>
+              Are you sure you want to remove{" "}
+              <b>
+                {selectedUser.firstName} {selectedUser.lastName}
+              </b>{" "}
+              as a housing locator? They will still have access as a referring staff after removal.
+            </WarningMessageWrapper>
+            <ButtonsWrapper>
+              <ConfirmDelete
+                kind="secondary"
+                onClick={() => {
+                  setPopup(false);
+                }}
+              >
+                Cancel
+              </ConfirmDelete>
+
+              <ConfirmDelete
+                kind="primary"
+                onClick={() => {
+                  handleDemote(selectedUser);
+                  setPopup(false);
+                  setSucessfulDeletionPopup(true);
+                }}
+              >
+                Demote
+              </ConfirmDelete>
+            </ButtonsWrapper>
+          </Modal>
+        </>
+      )}
+
+      {sucessfulDeletionPopup && (
+        <>
+          <Overlay />
+          <Modal>
+            <XWrapper>
+              <XButton
+                onClick={() => {
+                  setSucessfulDeletionPopup(false);
+                }}
+              >
+                &times;
+              </XButton>
+            </XWrapper>
+            <DoneMessageHeaderWrapper>
+              <DoneMessageHeader>
+                <img src="dark_green_check.svg" alt="Complete" width="78px" height="78px" />
+                <WarningMessageWrapper>
+                  <b></b> has been removed as a housing locator.
```

Comment:
I think we still need the name variables?

<img width="606" alt="image" src="https://github.com/TritonSE/USHS-Housing-Portal/assets/24444266/4099e746-62f4-4d44-aa42-e73543e012f5">

Reply:
done!